Failing test_enum_class with Python 3.13.0a5

### Describe the bug

I try to build sphinx in Fedora Linux 41 with Python 3.13.0a5.
`test_enum_class` fails.
repr of the Enum class has changed. Now there's `names=<not given>` instead of None. This is an expected change, see: https://github.com/python/cpython/pull/116508#issuecomment-2015319193

```
_______________________________ test_enum_class ________________________________

app = <SphinxTestApp buildername='html'>

    @pytest.mark.sphinx('html', testroot='ext-autodoc')
    def test_enum_class(app):
        options = {"members": None}
        actual = do_autodoc(app, 'class', 'target.enums.EnumCls', options)
    
        if sys.version_info[:2] >= (3, 12):
            args = ('(value, names=None, *values, module=None, '
                    'qualname=None, type=None, start=1, boundary=None)')
        elif sys.version_info[:2] >= (3, 11):
            args = ('(value, names=None, *, module=None, qualname=None, '
                    'type=None, start=1, boundary=None)')
        else:
            args = '(value)'
    
>       assert list(actual) == [
            '',
            '.. py:class:: EnumCls' + args,
            '   :module: target.enums',
            '',
            '   this is enum class',
            '',
            '',
            '   .. py:method:: EnumCls.say_goodbye()',
            '      :module: target.enums',
            '      :classmethod:',
            '',
            '      a classmethod says good-bye to you.',
            '',
            '',
            '   .. py:method:: EnumCls.say_hello()',
            '      :module: target.enums',
            '',
            '      a method says hello to you.',
            '',
            '',
            '   .. py:attribute:: EnumCls.val1',
            '      :module: target.enums',
            '      :value: 12',
            '',
            '      doc for val1',
            '',
            '',
            '   .. py:attribute:: EnumCls.val2',
            '      :module: target.enums',
            '      :value: 23',
            '',
            '      doc for val2',
            '',
            '',
            '   .. py:attribute:: EnumCls.val3',
            '      :module: target.enums',
            '      :value: 34',
            '',
            '      doc for val3',
            '',
        ]
E       AssertionError: assert ['', '.. py:c...ass', '', ...] == ['', '.. py:c...ass', '', ...]
E         At index 1 diff: '.. py:class:: EnumCls(value, names=<not given>, *values, module=None, qualname=None, type=None, start=1, boundary=None)' != '.. py:class:: EnumCls(value, names=None, *values, module=None, qualname=None, type=None, start=1, boundary=None)'
```

### How to Reproduce

Probably by running tests with Python 3.13.0a5, but I haven't tried it on the upstream code.

### Environment Information

```text
Sphinx 7.2.6
```


### Sphinx extensions

_No response_

### Additional context

_No response_

--- a/tests/test_extensions/test_ext_autodoc.py
+++ b/tests/test_extensions/test_ext_autodoc.py
@@ -1454,7 +1454,7 @@ class _EnumFormatter:
         """Generate the brief part of the class being documented."""
         assert doc, f'enumeration class {self.target!r} should have an explicit docstring'
 
-        if sys.version_info[:2] >= (3, 13) or sys.version_info >= (3, 12, 3):
+        if sys.version_info[:2] >= (3, 13) or sys.version_info[:3] >= (3, 12, 3):
             args = ('(value, names=<not given>, *values, module=None, '
                     'qualname=None, type=None, start=1, boundary=None)')
         elif sys.version_info[:2] >= (3, 12):

--- a/tests/test_extensions/test_ext_autodoc.py
+++ b/tests/test_extensions/test_ext_autodoc.py
@@ -1454,7 +1454,7 @@ class _EnumFormatter:
         """Generate the brief part of the class being documented."""
         assert doc, f'enumeration class {self.target!r} should have an explicit docstring'
 
-        if sys.version_info[:2] >= (3, 13):
+        if sys.version_info[:2] >= (3, 13) or sys.version_info >= (3, 12, 3):
             args = ('(value, names=<not given>, *values, module=None, '
                     'qualname=None, type=None, start=1, boundary=None)')
         elif sys.version_info[:2] >= (3, 12):

